MapboxNavigation/SpeechSynthesizerController.swift

```diff
@@ -0,0 +1,85 @@
+
+import Foundation
+import MapboxDirections
+import MapboxCoreNavigation
+
+public protocol SpeechSynthesizerController: class {
+    typealias SpeechSynthesizerCompletion = (Error?) -> Void
+    
+    ///
+    var delegate: SpeechSynthesizerDelegate? { get set }
+    
+    ///
+    var muted: Bool { get set }
+    ///
+    var volume: Float { get set }
+    ///
+    var isSpeaking: Bool { get }
+    ///
+    var locale: Locale { get set }
+    
+    ///
+    func changedIncomingSpokenInstructions(_ instructions: [SpokenInstruction])
+    ///
+    func speak(_ instruction: SpokenInstruction, completion: SpeechSynthesizerCompletion?)
+    
+    ///
+    func stopSpeaking()
+    ///
+    func interruptSpeaking() // ??
```

Comment:
Not sure if we need to, but maybe it could be useful to have different ways to shut down ongoing phrase? For example: one method to mark that phrase should be stopped gracefully whenever appropriate, and another to abort it immediately.